library_builders.sh

```diff
@@ -91,21 +92,26 @@ function build_adios2 {
     if [ "$(uname -s)" = "Linux" ]
     then
         EVPATH_ZPL="ON"
+        # ADIOS 2.7.1 & Blosc 1.20.1/1.21.0
+        #   /usr/local/lib/libblosc.a(blosc.c.o): In function `blosc_init.part.9':
+        # blosc.c:(.text+0x43e): undefined reference to `pthread_atfork'
```

Comment:
same problem appears again in my env on summit at import time...

Reply:
Maybe we can try (here and for blosc below):
```
export LDFLAGS="-lpthread -lpthread_nonshared"
```
https://gitlab.kitware.com/cmake/cmake/-/issues/20709#note_760439

Reply:
I think this line in blosc is outdated:
https://github.com/Blosc/c-blosc/blob/7435f28dd08606bd51ab42b49b0e654547becac4/blosc/CMakeLists.txt#L68
Is: `CMAKE_THREAD_PREFER_PTHREAD` ([docs](https://cmake.org/cmake/help/v3.2/module/FindThreads.html))
Should be: `THREADS_PREFER_PTHREAD_FLAG` ([docs](https://cmake.org/cmake/help/latest/module/FindThreads.html))

Reply:
started a draft in #1013